dadi/lib/handlers/image.js

```diff
@@ -966,6 +978,36 @@ ImageHandler.prototype.process = function (imageBuffer, options, imageInfo) {
   })
 }
 
+/**
+ * Transcodes an input buffer to a GIF, ensures colours are appropriate
+ * for GIF encoding via a "quantize" method.
+ *
+ * Saves the encoded GIF to a temporary file and removes to before returning
+ * the buffer.
+ *
+ * @param {Buffer} buffer - a Buffer extracted from the main image
+ * processor after applying image manipulations
+ * @returns {Buffer} a GIF encoded buffer
+ */
+ImageHandler.prototype.processGif = function (buffer) {
+  return new Promise((resolve, reject) => {
+    Jimp.read(buffer).then(image => {
+      let bitmap = new BitmapImage(image.bitmap)
+      GifUtil.quantizeDekker(bitmap)
+      let frame = new GifFrame(bitmap)
+
+      let tmpGifFile = `${path.join(tmpDirectory, sha1(this.parsedUrl.original.path))}.gif`
+
+      GifUtil.write(tmpGifFile, [frame]).then(gif => {
+        fs.unlinkSync(tmpGifFile)
```

Comment:
I would make this async, if possible. If you want to make it use Promises without any extra work, you could replace `fs` with `fs-extra` and do:

```js
return fs.unlink(tmpGifFile).then()
```